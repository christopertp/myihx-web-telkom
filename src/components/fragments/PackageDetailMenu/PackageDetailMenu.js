import React from 'react';
import PropTypes from 'prop-types';
import { useSelector } from 'react-redux';
import { Link } from 'react-router-dom';
import Button from '../../elements/Button';
import Spinner from '../../elements/Spinner';
import { thousand } from '../../../utils/format';
import styles from './styles.scoped.css';

export default function PackageDetailMenu(props) {
  const { onClickSubscribe, onClickDetail } = props;
  const { isLoading, data } = useSelector(s => s.packageDetail);
  const { productInfo } = data;

  return (
    <header className={styles.root}>
      <section>
        <Link to="/shop/internet">
          <img alt="back white" src="/assets/ic_back_white.svg" />
          Shop / Internet package
        </Link>
        <div>
          <h1>{data.productName}</h1>
          <p>{data.productDescription}</p>
          <Link to="">
            Learn more about internet speed
            <img alt="chevron white" src="/assets/ic_chevron_white.svg" />
          </Link>
          <aside>
            <h3>Rp{thousand(data.discountPrice)}<small> /month</small></h3>
            <Link onClick={onClickDetail} to="#">
              View Pricing Detail
              <img alt="chevron red" src="/assets/ic_chevron_red.svg" />
            </Link>
            <Button disabled={isLoading.submit} onClick={onClickSubscribe}>
              {isLoading.submit ? <Spinner /> : 'Subscribe'}
            </Button>
          </aside>
        </div>
      </section>
      <ul>
        {
          isLoading.data
            ? [...Array.from({ length: 2 }).keys()].map(i => <li className="loading" key={i} />)
            : productInfo.map((i, idx) =>(
              <li key={idx}>
                <aside>
                  <h3>{i.value}</h3>
                  <span>{i.unit}</span>
                </aside>
                <h5>{i.name}</h5>
                <span>{i.description}</span>
              </li>
            ))
        }
      </ul>
    </header>
  );
}

PackageDetailMenu.defaultProps = {
  onClickDetail: () => { },
  onClickSubscribe: () => { },
};

PackageDetailMenu.propTypes = {
  onClickDetail: PropTypes.func,
  onClickSubscribe: PropTypes.func,
};
