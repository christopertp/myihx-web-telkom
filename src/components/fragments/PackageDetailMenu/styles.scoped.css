.root { z-index: 2; }

.root > section {
  background: #ee3124 url(/assets/bg_product_internet.jpg) no-repeat center;
  background-size: cover;
  margin: -4rem var(--neg-side-margin) -3rem;
  padding: 3rem var(--side-margin) 6rem;
}

.root > section > a {
  color: rgba(255, 255, 255, 0.75);
  font-weight: 800;
  text-transform: uppercase;
  display: flex;
  align-items: center;
  font-size: .9rem;
  margin-bottom: 3rem;
}

.root > section > a > img { height: 0.9rem; }

.root > section > div {
  display: grid;
  grid-template-columns: 1fr auto;
  grid-template-rows: auto auto auto;
}

.root > section > div > h1:empty {
  background-color: #db1e11;
  height: 3.75rem;
  opacity: 0.6;
  width: 23rem;
}

.root > section > div > p:empty {
  background-color: #db1e11;
  height: 1.5rem;
  opacity: 0.6;
  width: 30rem;
}

.root > section > div > h1 {
  align-self: end;
  color: #ffffff;
  font-size: 2.5rem;
}

.root > section > div > p {
  align-self: center;
  color: rgba(255, 255, 255, 0.75);
  grid-column: 1;
  margin: 0.5rem 0 1rem;
}

.root > section > div > a {
  color: #ffffff;
  font-weight: 800;
  text-transform: uppercase;
}

.root > section > div > a > img {
  height: 1rem;
  vertical-align: middle;
}

.root > section > div > aside {
  background-color: #ffffff;
  border-radius: 0.5rem;
  box-shadow: 0 0.5rem 1rem rgba(0, 0, 1, 0.15);
  grid-column: 2;
  grid-row: 1/4;
  justify-self: end;
  margin: auto 0 auto 1rem;
  padding: 1rem;
  width: 16.375rem;
}

.root > section > div > aside > h3 {
  font-size: 1.5rem;
  line-height: 1.2;
}

.root > section > div > aside > h3 > small {
  color: #999999;
  font-size: 1rem;
  font-weight: 400;
}

.root > section > div > aside > a {
  align-items: center;
  color: #ee3124;
  display: flex;
  margin: 0.5rem 0 1rem;
}

.root > section > div > aside > a > img { height: 1rem; }

.root > section > div > aside > button { width: 100%; }

.root > ul { display: flex; }

.root > ul > li:empty {
  background-color: #dddbdd;
  box-shadow: none;
  height: 7.75rem;
}

.root > ul > li {
  background-color: #ffffff;
  border-radius: 0.5rem;
  box-shadow: 0 0.25rem 0.5rem rgba(0, 0, 1, 0.075);
  display: grid;
  flex: 1;
  grid-template-columns: auto 1fr;
  grid-template-rows: auto 1fr;
  padding: 1rem;
}

.root > ul >li:nth-child(2) { margin: 0 1.25rem; }

.root > ul > li > aside {
  align-items: center;
  border-radius: 0.5rem;
  display: flex;
  flex-direction: column;
  grid-row: 1/3;
  justify-content: center;
  height: 5.375rem;
  margin-right: 1rem;
  width: 5.375rem;
}

.root > ul > li:first-child > aside { background: linear-gradient(135deg, #6a120c, #b21f15, #ee4c24, #ff7e00); }
.root > ul > li:nth-child(2) > aside { background: linear-gradient(135deg, #080c10, #115b79); }
.root > ul > li:last-child > aside { background: linear-gradient(135deg, #284050, #0eb37e); }

.root > ul > li > aside > h3 {
  color: #ffffff;
  font-size: 2rem;
  font-weight: 700;
}

.root > ul > li > aside > span {
  color: rgba(255, 255, 255, 0.5);
  font-size: 0.8rem;
  text-transform: uppercase;
}

.root > ul > li > span {
  color: #000001;
  grid-column: 2;
}

@media only screen and (max-width: 48em) {
  .root {
    background-color: #f2f2f2;
    margin: 0 -1rem;
  }
  .root > section {
    margin: -1rem 0;
    padding: 1rem 1rem 2rem;
  }
  .root > section > a {
    color: transparent;
    font-size: 0;
    margin-bottom: 1rem;
  }
  .root > section > a > img { height: 1.5rem; }
  .root > section > div { display: block; }
  .root > section > div > h1:empty { width: 50%; }
  .root > section > div > p:empty { width: 100%; }
  .root > section > div > h1 { 
    font-size: 1.75rem;
    line-height: 1.2;
  }
  .root > section > div > p { font-size: 0.85rem; }
  .root > section > div > a { font-size: 0.9rem; }
  .root > section > div > aside { display: none }
  .root > ul {
    flex-direction: column;
    padding: 0 1rem;
  }
  .root > ul > li:empty { height: 6.5rem; }
  .root > ul > li {
    grid-template-rows: 2fr 3fr;
    margin-bottom: 0.5rem;
  }
  .root > ul > li:nth-child(2) { margin: 0 0 0.5rem 0; }
  .root > ul > li > aside {
    height: 4.625rem;
    width: 4.625rem;
  }
  .root > ul > li > aside > h3 {
    font-size: 1.75rem;
  }
  .root > ul > li > h5 {
    align-self: end;
    font-size: 1rem;
    margin-bottom: 0.25rem;
  }
  .root > ul > li > span {
    font-size: 0.85rem;
    line-height: 1.4;
  }
}
