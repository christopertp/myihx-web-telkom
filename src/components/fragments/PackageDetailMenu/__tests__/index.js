import React from 'react';
import { useSelector } from 'react-redux';
import ShallowRenderer from 'react-test-renderer/shallow';
import Spinner from '../../../elements/Spinner';
import PackageDetailMenu from '../PackageDetailMenu';

useSelector.mockImplementation(fn => {
  fn({});
  return {
    data: {
      productInfo: [{ name: 'nama', description: 'desc' }],
    },
    isLoading: { data: false, submit: false },
  };
});

describe('src/components/fragments/PackageDetailMenu', () => {
  test('render', () => {
    PackageDetailMenu.defaultProps.onClickDetail();
    PackageDetailMenu.defaultProps.onClickSubscribe();
    const shallow = new ShallowRenderer();
    const tree = shallow.render(<PackageDetailMenu />);
    expect(tree).toMatchSnapshot();
  });

  test('loading state', () => {
    const isLoading = { data: true, submit: true };
    useSelector.mockImplementationOnce(() => ({ data: {}, isLoading }));
    const result = PackageDetailMenu(PackageDetailMenu.defaultProps);
    const ul = result.props.children[1];
    expect(ul.props.children.length).toBe(2);
    expect(ul.props.children[0].props.children).toBe(undefined);
    const aside = result.props.children[0].props.children[1].props.children[3];
    expect(aside.props.children[2].props.children.type).toBe(Spinner);
  });
});
