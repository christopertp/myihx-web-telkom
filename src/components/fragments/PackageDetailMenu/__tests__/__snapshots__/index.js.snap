// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/components/fragments/PackageDetailMenu render 1`] = `
<header>
  <section>
    <Link
      to="/shop/internet"
    >
      <img
        alt="back white"
        src="/assets/ic_back_white.svg"
      />
      Shop / Internet package
    </Link>
    <div>
      <h1 />
      <p />
      <Link
        to=""
      >
        Learn more about internet speed
        <img
          alt="chevron white"
          src="/assets/ic_chevron_white.svg"
        />
      </Link>
      <aside>
        <h3>
          Rp
          0
          <small>
             /month
          </small>
        </h3>
        <Link
          onClick={[Function]}
          to="#"
        >
          View Pricing Detail
          <img
            alt="chevron red"
            src="/assets/ic_chevron_red.svg"
          />
        </Link>
        <Button
          className=""
          disabled={false}
          onClick={[Function]}
          to=""
          type="button"
          variant="red"
        >
          Subscribe
        </Button>
      </aside>
    </div>
  </section>
  <ul>
    <li>
      <aside>
        <h3 />
        <span />
      </aside>
      <h5>
        nama
      </h5>
      <span>
        desc
      </span>
    </li>
  </ul>
</header>
`;
